cpp/CMakeLists.txt

```diff
@@ -160,7 +160,7 @@ if(CMAKE_COMPILER_IS_GNUCXX)
 endif(CMAKE_COMPILER_IS_GNUCXX)
 
 set(CMAKE_CUDA_FLAGS
-  "${CMAKE_CUDA_FLAGS} -Xcudafe --diag_suppress=unrecognized_gcc_pragma")
+  "${CMAKE_CUDA_FLAGS} -Xcudafe --diag_suppress=unrecognized_gcc_pragma --expt-relaxed-constexpr")
```

Comment:
why do you need this additional constexpr flag here?

Reply:
This isn't needed for this particular PR but it allows us to call constants from the std libraries directly in CUDA kernels. If not merged with this PR, it will be made when we move the sparse prims over.